### DISCUSSION TITLE
Day of Reset and Scheduled Annoucements

### DISCUSSION THREAD

**Discussion by njhughes-01** [Diyagi/vrising-server-docker#7] [Help] [answered]
I am assuming the ```VR_DAY_OF_RESET``` and ```VR_RESET_DAYS_INTERVAL``` are referring to wipe days. 

How do you think the best way to handle normal daily server resets to pull hotfixes and such. 
I also was looking at a way to automate some announcement msgs about the server discord link scheduling, things like that. 
Do you think the best approach would be to rcon using cron on the host machine, or use something internal to the container? 
I wanted your opinion before I went any direction.

**Comment by Diyagi** [comment]
Auto restart could be added inside the container as cron, probably using [Supercronic](https://github.com/aptible/supercronic).
For the announcement messages you could use Host cron with RCON for now, since i have no ideia how to make these inside the container while still being easy to configure.

**Comment by njhughes-01** [comment]
It makes sense to me just to manage these on the host.  I made a service file for the container to launch on boot  and to keep alive
```
[Unit]
Description=Docker Compose Service for vRising
Requires=docker.service
After=docker.service

[Service]
Type=oneshot
RemainAfterExit=true
WorkingDirectory=/mnt/share/Backup/steamHosting/
ExecStart=/usr/bin/docker-compose up -d
ExecStop=/usr/bin/docker-compose down

[Install]
WantedBy=multi-user.target
```
```
0 6 * * * /usr/local/bin/docker-compose -f /mnt/share/Backup/steamHosting/docker-compose.yml down
2 6 * * * /usr/local/bin/docker-compose -f /mnt/toth/Backup/steamHosting/docker-compose.yml restart
```
I'll write up a script to handle this more gracefully and add-in announcements before the server down. Let me know if you would like me to add any of this information or if you think I am way off base here .

**Comment by njhughes-01** [comment]
OK so I have most of this going on the host but i think we could add 
rcon-cli (https://github.com/gorcon/rcon-cli/releases/tag/v0.10.3)

Then this could all be managed with cron inside the container, using all env var to populate the rcon.yaml and leave it in persistent data. 

It would still require the user to docker exec into it to change the cron

I can make up a pr for this if it is something that is wanted

**Reply by Diyagi** [reply]
I added this about an hour ago, should be working after you repull the image.

**Reply by Diyagi** [reply, marked as answer]
[RCON](https://github.com/Diyagi/vrising-server-docker?tab=readme-ov-file#rcon-commands)

**Comment by njhughes-01** [comment]
quick work. Thanks here are a couple of dirty scripts  i made up to manage messaging and restart

serverRestart.sh
```
#!/bin/bash

# Function to announce server restart with countdown
announce_restart() {
    local MINUTES="$1"
    /usr/local/bin/rcon "announcerestart $MINUTES"
    for ((i=MINUTES-1; i>=1; i--)); do
        /usr/local/bin/rcon "announce Server restart in $i minutes"
        sleep 60 # Sleep for 1 minute
    done
}

# Check if the number of minutes is provided as an argument
if [ $# -ne 1 ]; then
    echo "Usage: $0 <number of minutes until restart>"
    exit 1
fi

# Get the number of minutes from the command line argument
MINUTES="$1"

# Announce server restart with countdown
announce_restart "$MINUTES"

# Shutdown Pull And Start Server
./restartPull.sh
```
restartPull.sh
```
#!/bin/bash

# Change directory to where your docker-compose.yml file is located
cd /mnt/toth/Backup/steamHosting/

# Stop the containers
/usr/local/bin/docker-compose down

# Check if docker-compose down was successful
if [ $? -eq 0 ]; then
    # Start the containers with pulling latest images
    /usr/local/bin/docker-compose up -d --pull
else
    echo "Error: Failed to stop containers. Exiting."
    exit 1
fi
```
discordAnnounce.sh

```
#!/bin/bash

# Check if the environment variable is set
if [ -z "$DISCORD_LINK" ]; then
    echo "DISCORD_LINK environment variable is not set. Please set it with the Discord invite link."
    exit 1
fi

# Use the environment variable value for the link
LINK="$DISCORD_LINK"

# Execute the command with the link
/usr/local/bin/rcon "announce Join Us On Discord: $LINK"
```
SET Discord line with ENV DISCORD_LINK

Saving these as scripts in persistent data and updating the paths might be helpful to someone in future. 


Currently they are designed to run on host but could be moved into our container with the rcon update

**Reply by Diyagi** [reply]
Even tho theres an announcerestart command, i feel like shutdown is better, since you can do something like this `shutdown 10,5,3,2,1 Server will restart in ~{t}m. Reason: Scheduled Restart` and the game will take care of everything else.

Im working in the auto update and auto restart right now

**Comment by Diyagi** [comment]
Auto Update, Auto Reboot and Auto Announce are all added to the image.
You should also check [Formatting announce messages](https://github.com/Diyagi/vrising-server-docker/wiki/Formatting-announce-messages)

**Reply by njhughes-01** [reply]
Wow that's pretty Amazing stuff great work
